py/desispec/scripts/proc_joint_fit.py

```diff
@@ -508,22 +530,25 @@ def main(args=None, comm=None):
                 cmd += " --maxstdstars {}".format(args.maxstdstars)
 
             inputs = framefiles[sp] + skyfiles[sp] + fiberflatfiles[sp]
-            num_cmd += 1
             if subcomm is None:
                 #- Using multiprocessing
+                num_cmd += 1
                 err = runcmd(cmd, inputs=inputs, outputs=[stdfile])
             else:
                 #- Using MPI
                 try:
                     cmdargs = cmd.split()[1:]
                     cmdargs = desispec.scripts.stdstars.parse(cmdargs)
+                    if subcomm.rank == 0:
+                        num_cmd += 1
                     err = runcmd(desispec.scripts.stdstars.main, 
-                        args=(cmdargs, subcomm), inputs=inputs, outputs=[stdfile]
+                        args=cmdargs, inputs=inputs, outputs=[stdfile], comm=subcomm
                     )
                 except:
                     #- Catches sys.exit from stdstars.main
                     log.error('stdstars.main failed for {}'.format(os.path.basename(stdfile)))
                     err = True
+
             if err:
                 num_err += 1
```

Comment:
For the case of `srun -n 30 ... desi_proc_joint_fit ...`:
Since `num_cmd` is only incremented for `subcomm.rank==0`, but `num_err` is incremented for every rank, if `runcmd` fails for all ranks this leads to `num_cmd==10` and `num_err==30`, which causes this later check to fail to recognize it as a failure:
```
        if num_err>0 and num_err==num_cmd:
            if rank == 0:
                log.critical('All stdstar commands failed')
            sys.exit(1)
```
and it doesn't properly exit with an error to let slurm know that the job failed.

Reply:
duplicate